maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,217 @@
+
+class RemoveImapTags extends Maintenance {
+	static protected $dryRun = false;
+
+	private $maps;
+	private $mapsClientConfig;
+	private $slaveDBConfig;
+
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Removes from articles <imap /> tags for given tiles' set id.";
+		$this->addOption( 'dry-run', 'Dry run; make no changes', $required = false, $withArg = false );
+		$this->addOption(
+			'tiles-set-id',
+			'Specify a tiles\' set id which leads to a group of maps which <imap /> tags should get removed',
+			$required = true,
+			$withArg = true,
+			'tsi'
+		);
+	}
+
+	static public function isDryRun() {
+		return self::$dryRun;
+	}
+
+	static public function isNaturalNumber( $int ) {
+		return intval( $int ) > 0;
+	}
+
+	public function getMapsIdsUsingTileset( $cityId, $tilesSetId ) {
+		$dbw = new DatabaseMysqli(
+			$this->slaveDBConfig['host'],
+			$this->slaveDBConfig['user'],
+			$this->slaveDBConfig['password'],
+			$this->slaveDBConfig['database']
+		);
+
+		return $dbw->selectFieldValues( 'map', 'id', [
+				'city_id' => $cityId,
+				'tile_set_id' => $tilesSetId
+		] );
+	}
+
+	public function getArticlesIdsUsingImap( $cityId ) {
+		global $wgStatsDB;
+
+		$dbw = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
+		return $dbw->selectFieldValues( 'city_used_tags', 'ct_page_id', [
+			'ct_wikia_id' => $cityId,
+			'ct_kind' => 'imap'
+		] );
+	}
+
+	public function hasImapTag( $articleId, &$foundTags, &$foundTagsMapIds ) {
+		$article = Article::newFromID( $articleId );
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$results = null;
+			$articleContent = $article->getContent();
+			$imapSearchRegexp = "/<imap.*map\-id\=['\"](\d{1,})['\"].*(<\/imap|\/)>/";
+
+			$noOfFoundTags = preg_match_all($imapSearchRegexp, $articleContent, $results);
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function hasTagsToRemove( $foundTagsMapIds, $mapsUsingTheTileset, &$tagsToRemoveArrayKeyIds ) {
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
+			}
+		}
+
+		return !empty( $tagsToRemoveArrayKeyIds );
+	}
+
+	public function removeImapTagFromArticle( $articleId, $stringWithTagToRemove ) {
+		$user = User::newFromName( 'WikiaBot' );
+		$article = Article::newFromID( $articleId );
+
+		if ( $user->getId() === 0 ) {
+			$this->output( 'Failed using WikiaBot user' . PHP_EOL );
+		}
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$oldContent = $article->getContent();
+			$newContent = str_replace( $stringWithTagToRemove, "", $oldContent );
+
+			$this->output( sprintf( 'Trying to edit article #%d', $articleId ) . PHP_EOL );
+			if( !self::isDryRun() ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)